app/src/main/java/io/neurolab/Filter.java

```diff
@@ -0,0 +1,65 @@
+package io.neurolab;
+
+public class Filter {
+
+    private double resonance;
+    private double frequency;
+    private int sampleRate;
+    private boolean lowPass;
+    private double c, a1, a2, a3, b1, b2;
+    private double[] inputHistory = new double[2];
+    private double[] outputHistory = new double[3];
+
+    public Filter() {
+    }
+
+    public Filter(double frequency, int sampleRate, boolean lowPass, double resonance) {
+        setFilter(frequency, sampleRate, lowPass, resonance);
+    }
+
+    private void setCoefficients() {
+        if (lowPass) {
+            c = 1.0f / (float) Math.tan(Math.PI * frequency / sampleRate);
+            a1 = 1.0f / (1.0f + resonance * c + c * c);
+            a2 = 2f * a1;
+            a3 = a1;
+            b1 = 2.0f * (1.0f - c * c) * a1;
+            b2 = (1.0f - resonance * c + c * c) * a1;
```

Comment:
Check values with desktop app and enclose them in brackets to improve readability. (as an example, rather than a + b * c + d * e ==> (a + b) * (c + d) * e)